packages/teleport-types/src/uidl.ts

```diff
@@ -19,18 +19,54 @@ export interface UIDLGlobalProjectValues {
   manifest?: WebManifest
   variables?: Record<string, string>
 }
-export interface UIDLGlobalAsset {
-  type: 'script' | 'style' | 'font' | 'canonical' | 'icon'
-  path?: string
-  content?: string
+export interface UIDLScriptExternalAsset {
+  type: 'script'
+  path: string
   options?: {
     async?: boolean
     defer?: boolean
     target?: string
+  }
+}
+export interface UIDLScriptInlineAsset {
+  type: 'script'
+  content: string
+}
```

Comment:
```suggestion
export interface UIDLScriptExternalAsset {
  type: 'script'
  path: string
  options?: {
    async?: boolean
    defer?: boolean
    target?: string
  }
}
export interface UIDLScriptInlineAsset {
  type: 'script'
  content: string,
  options?: {
    async?: boolean
    defer?: boolean
    target?: string
  }
}
```

It's just that we can't have both content and path at the same time 😄